Calculator_Project/Source.cpp

```diff
@@ -0,0 +1,251 @@
+#include <iostream>
+#include <vector>
+#include <iterator>
+#include <string>
+
+template <typename T>
+void vec_print(const std::vector<T>& container) {   //Print vector, entry by entry
+    for (const T equation : container)              //Found online, modified to generic type
+        std::cout << equation << ' ';
+    std::cout << std::endl;
+}
+
+bool char_is_num(const char c) {        //Return true if character c is a numeral, false if not
+    if ((c == '0') or (c == '1') or (c == '2') or (c == '3') or (c == '4') or (c == '5') or (c == '6') or (c == '7') or (c == '8') or (c == '9')) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+
+int main() {
+
+    std::string equ_str;                    //Initialize equation variables, and take input of equation
+    std::cout << "Enter equation: ";
+    std::cin >> equ_str;
+    std::vector<std::string> equ_vec;
+
+    std::string num_str;                    //Variable to hold numeric characters, to combine multiple numeric characters into one number
+    int equ_str_index = 0;                  //Variable to track which character in the equation we are one
+    for (char c : equ_str) {                //Iterate through the equation, one character at a time
+
+        switch (c) {
+        case '0':                           //If the current character is a 0;
+            if (equ_str_index == 0) {       //If the current character is the first in the string, or if the preceding character is not a number,
+                num_str = "0";              //the current character is the first character in a number-string, so set num_str equal to the character, typed as a string
+            }
+            else if (char_is_num(equ_str[equ_str_index-1]) == false) {
+                num_str = "0";
+            }
+            else {                          //If the current character is not the first in the string, and is preceded by a number,
+                num_str.append("0");        //it is part of an existing number-string, so append it to the existing num_str
+            }
+
+            if (equ_str_index == (equ_str.length() - 1)) {                  //If the current character is the last in the equation,
+                equ_vec.push_back(num_str);                                 //or the next character is not a number,
+            }                                                               //the current character is the last character in the number-string,
+            else if (char_is_num(equ_str[equ_str_index+1]) == false) {      //so we append num_str to the end of the equation vector.
+                equ_vec.push_back(num_str);                                 //If it isn't the last character in the number-string, don't do anything
+            }
+            break;

```

Comment:
We shouldn't have to repeat this code 10 times when the only thing that changes is the number that it appends to the string

Reply:
We could simplify it by turning the `char c` we look at into a string, like `std::string digit { c };` and then replacing the whole switch statement with 
```c
std::string digit { c };
if (equ_str_index == 0) {
    num_str = digit;
} else if (char_is_num(equ_str[equ_str_index-1]) == false) {
    num_str = digit;
} else {
    num_str.append(digit);
}

if (equ_str_index == (equ_str.length() - 1)) {
    equ_vec.push_back(num_str);
} else if (char_is_num(equ_str[equ_str_index+1]) == false) { 
    equ_vec.push_back(num_str);
}
```